catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/FederationAdminServiceImpl.java

```diff
@@ -66,147 +131,867 @@ public FederationAdminServiceImpl() {
         this.security = security;
     }
 
-    public void init() {
-
-    }
-
-    public void setCatalogFramework(CatalogFramework catalogFramework) {
-        this.catalogFramework = catalogFramework;
+    @Override
+    public String addRegistryEntry(String xml) throws FederationAdminException {
+        Metacard metacard = getRegistryMetacardFromString(xml);
+        return addRegistryEntry(metacard);
     }
 
-    public void setFilterBuilder(FilterBuilder filterBuilder) {
-        this.filterBuilder = filterBuilder;
+    @Override
+    public String addRegistryEntry(String xml, Set<String> destinations)
+            throws FederationAdminException {
+        Metacard metacard = getRegistryMetacardFromString(xml);
+        return addRegistryEntry(metacard, destinations);
     }
 
-    public void setParser(Parser parser) {
-        this.configurator =
-                parser.configureParser(Arrays.asList(RegistryObjectType.class.getPackage()
-                                .getName(),
-                        net.opengis.ogc.ObjectFactory.class.getPackage()
-                                .getName(),
-                        net.opengis.gml.v_3_1_1.ObjectFactory.class.getPackage()
-                                .getName()), FederationAdminServiceImpl.class.getClassLoader());
-        this.configurator.addProperty(Marshaller.JAXB_FRAGMENT, true);
-        this.parser = parser;
+    @Override
+    public String addRegistryEntry(Metacard metacard) throws FederationAdminException {
+        return addRegistryEntry(metacard, null);
     }
 
-    public void setRegistryTransformer(InputTransformer inputTransformer) {
-        this.registryTransformer = inputTransformer;
+    @Override
+    public String addRegistryEntry(Metacard metacard, Set<String> destinations)
+            throws FederationAdminException {
+        validateRegistryMetacard(metacard);
+
+        String metacardId;
+
+        Subject systemSubject = security.getSystemSubject();
```

Comment:
Going the route that @pklinef is suggesting is definitely much safer. However, if all of the interactions with this service are expected to be from some sort of user facing endpoint, there should always be a user subject available. You should just be able to use the subject that is available and allow all of the catalog plugins to check if the user has authorization to do these registry actions.